src/lib/components/Widget.tsx

```diff
@@ -115,7 +104,7 @@ export default function Widget(props: PropsWithChildren<WidgetProps>) {
                   <AtomProvider>
                     <Web3Provider provider={provider} jsonRpcEndpoint={jsonRpcEndpoint}>
                       <Updaters />
-                      <ErrorGenerator {...props} />
+                      <PropValidator {...props} />
```

Comment:
Can propValidator be a hook instead of a component? It's not rendered. Is there an advantage to having a component instead of a hook?

Reply:
Yeah, rendering it in the component tree here causes the errors it throws to be caught by the error boundary above.

Reply:
This makes sense for WidgetPropsValidator, but couldn't SwapPropsValidator still be a hook?